src/Nethermind/Nethermind.Consensus.AuRa/Contracts/RewardContract.cs

```diff
@@ -117,11 +111,11 @@ public RewardContract(IAbiEncoder abiEncoder, Address contractAddress, long tran
         /// 101-106 - Uncle - Reward attributed to uncles, with distance 1 to 6 (Ethash engine)
         /// </param>
         public Transaction Reward(Address[] benefactors, ushort[] kind)
-            => GenerateTransaction(_abiEncoder.Encode(Definition.reward, benefactors, kind), Address.SystemUser);
+            => GenerateSystemTransaction(_abiEncoder.Encode(Definition.Functions[RewardFunction].GetCallInfo(), benefactors, kind));
         
         public (Address[] Addresses, BigInteger[] Rewards) DecodeRewards(byte[] data)
         {
-            var objects = _abiEncoder.Decode(Definition.rewardReturn, data);
+            var objects = _abiEncoder.Decode(Definition.Functions[RewardFunction].GetReturnInfo(), data);
```

Comment:
you could potentially use type converters and generic parameters in the Decode method

Reply:
Would have to use multiple signatures with multiple tuple types (generating?) too much hussle for now.